config/initializers/secure_headers.rb

```diff
@@ -14,11 +12,17 @@
   config.x_content_type_options = "nosniff"
 
   # http://msdn.microsoft.com/en-us/library/dd565647%28v=vs.85%29.aspx
-  config.x_xss_protection = { :value => 1 }
+  config.x_xss_protection = "1; mode=block"
 
   # https://w3c.github.io/webappsec/specs/content-security-policy/
-  config.csp = false
+  config.csp = {
+    # "meta" values. these will shaped the header, but the values are not included in the header.
+    report_only:  true,
+
+    # directive values: these values will directly translate into source directives
+    default_src: %w(https: 'self'),
+  }
 
   # https://www.nwebsec.com/HttpHeaders/SecurityHeaders/XDownloadOptions
-  config.x_download_options = false
+  config.x_download_options = nil
```

Comment:
(Set to `SecureHeaders::OPT_OUT` for now as those kind of discussions feel like they're out of scope for a gem upgrade task)